x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/unified/UnifiedChatCompletionRequestEntity.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.unified;
+
+import org.elasticsearch.inference.UnifiedCompletionRequest;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.external.http.sender.UnifiedChatInput;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class UnifiedChatCompletionRequestEntity implements ToXContentFragment {
+
+    public static final String NAME_FIELD = "name";
+    public static final String TOOL_CALL_ID_FIELD = "tool_call_id";
+    public static final String TOOL_CALLS_FIELD = "tool_calls";
+    public static final String ID_FIELD = "id";
+    public static final String FUNCTION_FIELD = "function";
+    public static final String ARGUMENTS_FIELD = "arguments";
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String PARAMETERS_FIELD = "parameters";
+    public static final String STRICT_FIELD = "strict";
+    public static final String TOP_P_FIELD = "top_p";
+    public static final String STREAM_FIELD = "stream";
+    private static final String NUMBER_OF_RETURNED_CHOICES_FIELD = "n";
+    public static final String MESSAGES_FIELD = "messages";
+    private static final String ROLE_FIELD = "role";
+    private static final String CONTENT_FIELD = "content";
+    private static final String MAX_COMPLETION_TOKENS_FIELD = "max_completion_tokens";
+    private static final String STOP_FIELD = "stop";
+    private static final String TEMPERATURE_FIELD = "temperature";
+    private static final String TOOL_CHOICE_FIELD = "tool_choice";
+    private static final String TOOL_FIELD = "tools";
+    private static final String TEXT_FIELD = "text";
+    private static final String TYPE_FIELD = "type";
+    private static final String STREAM_OPTIONS_FIELD = "stream_options";
+    private static final String INCLUDE_USAGE_FIELD = "include_usage";
+
+    private final UnifiedCompletionRequest unifiedRequest;
+    private final boolean stream;
+
+    public UnifiedChatCompletionRequestEntity(UnifiedChatInput unifiedChatInput) {
+        Objects.requireNonNull(unifiedChatInput);
+
+        this.unifiedRequest = unifiedChatInput.getRequest();
+        this.stream = unifiedChatInput.stream();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startArray(MESSAGES_FIELD);
+        {
+            for (UnifiedCompletionRequest.Message message : unifiedRequest.messages()) {
+                builder.startObject();
+                {
+                    if (message.content() == null) {
```

Comment:
This section is different from what's in main only in that we use an if-else here instead of a switch-case because the jdk version is older in 8.x